synapse/http/servlet.py

```diff
@@ -191,6 +197,32 @@ def parse_string_from_args(
             return default
 
 
+def parse_string_from_args(
```

Comment:
Please add a docstring and type hints. The default case in this seems wrong -- it looks like it should `return default`. I'm not really sure how this is different than `parse_string`?

Reply:
> Please add a docstring and type hints.

Added. The docs are essentially the same as `parse_string` which may be why they were omitted before.


---

> The default case in this seems wrong -- it looks like it should `return default`

The default case should fall through from `parse_strings_from_args` which we return.

---

This was a previous method that I am maintaining.

There are `_from_args` variants of all of these type of parsing functions in this file. For example `parse_integer` and `parse_integer_from_args`. They are redundant either way but we can refactor these out in another PR.

For strings, there is `parse_string`,  `parse_string_from_args` and now `parse_strings_from_args` to support multiple values being passed in like `?prev_event=abc&prev_event=foo`.

Reply:
Ah I didn't realize that this was old code that wasn't fully added.